src/client/components/login-component/index.js

```diff
@@ -0,0 +1,110 @@
+
+class LoginComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      'status': 'being',
+      'error': null,
+    };
+  }
+
+  handleLoginResponse(status, response) {
+    if (status === 400) {
+      this.handleFieldsError(response);
+    } else if (status === 403) {
+      this.handleMisMatch(response);
+    } else if (status === 500) {
+      this.handleUnknownError(response);
+    } else if (status === 200) {
+      window.location.href = '/feed';
+    }
+  }
+
+  handleFieldsError(response) {
+    let error = JSON.parse(response).errorType;
+    let errorMessage;
+    if (error === 'ContentType') {
+      errorMessage = 'Something went wrong, please try later!';
+      this.setState({'error': errorMessage});
+    } else if (error === 'FieldMissing') {
+      errorMessage = 'Don\'t try to hack our site!';
+      this.setState({'error': errorMessage});
+    } else if (error === 'WrongFormat') {
+      errorMessage = 'Don\'t try to hack our site!';
+      this.setState({'error': errorMessage});
+    }
+  }
+
+  handleMisMatch(response) {
+    if (response.errorType === 'MisMatch') {
+      let errorMessage = 'Username and password mismatch!';
+      this.setState({'error': errorMessage});
+    }
+  }
+
+  handleUnknownError(response) {
+    if (response.errorType === 'Unknown') {
+      let errorMessage = 'Unknown Error!';
+      this.setState({'error': errorMessage});
+    }
+  }
+
+  collectData(event) {
+    return {
+      'username': event.target.elements.namedItem('username').value,
+      'password': event.target.elements.namedItem('password').value,
+    };
+  }
+
+  // validateData(data) {
+  //   let reg = /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$/;
+  //   if (!reg.test(data.username)) {
+  //     this.setState({'isUsernameValid': true});
+  //   }
+  // }
+
+  sendData(data) {
+    let xhr = new XMLHttpRequest();
+    xhr.addEventListener('readystatechange', function() {
+      if (xhr.readyState === XMLHttpRequest.DONE) {
+        this.handleLoginResponse(xhr.status, xhr.response);
+      }
+    }.bind(this));
+    xhr.open('POST', '/api/login');
+    xhr.setRequestHeader('Accept', 'application/json');
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    this.setState({'status': 'loading'});
+    xhr.send(JSON.stringify(data));
+  }
+
+  submitLogin(event) {
+    event.preventDefault();
+    let LoginInfo = this.collectData(event);
+    // this.validateData(LoginInfo);
+    this.sendData(LoginInfo);
+  }
+
+  createNewAccount(event) {
+    window.location.href = '/signup';
+  }
+
+  render() {
+    return (
+      <main >
+        <h1 className='login-title'>Log in to Variscite</h1>
+        <LoginForm isLoading={this.state.status === 'loading'}
+          errorMessage={this.state.error}
+          onSubmit={this.submitLogin.bind(this)} />
+        <p className='or'>or</p>
+        <Link className='newAccount' to="/feed">Create New Account</Link>
```

Comment:
- do you really want to navigate the user to the feed if he/she clicks on the "Create New Account" button?
- you can use a the "text-transform: capitalize" css setting for this link, dont need to capitalize in js